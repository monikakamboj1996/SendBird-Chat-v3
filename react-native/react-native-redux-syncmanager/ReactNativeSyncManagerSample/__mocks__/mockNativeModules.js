jest.mock('NativeModules', () => ({
  UIManager: {
    RCTView: () => {}
  },
  RNGestureHandlerModule: {
    attachGestureHandler: jest.fn(),
    createGestureHandler: jest.fn(),
    dropGestureHandler: jest.fn(),
    updateGestureHandler: jest.fn(),
    State: {},
    Directions: {
      forceTouchAvailable: jest.fn()
    }
  },
  PlatformConstants: {
    forceTouchAvailable: false
  }
}));
