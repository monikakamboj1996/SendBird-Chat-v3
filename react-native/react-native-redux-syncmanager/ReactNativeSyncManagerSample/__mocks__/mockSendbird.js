const defaultUser = {
  userId: '123',
  nickname: 'test_nickname',
  profileUrl: 'test_profileUrl'
};
export let user = JSON.parse(JSON.stringify(defaultUser));
export function setUser(newUser) {
  user = newUser;
}
export function resetUser() {
  user = JSON.parse(JSON.stringify(defaultUser));
}

const defaultChannel = {
  channelUrl: 'url'
};
export function resetChannel() {
  channel = JSON.parse(JSON.stringify(defaultChannel));
}

export let channel = {
  channelUrl: 'url'
};

export function restoreAllDefaultMocks() {
  mockUpdateCurrentUserInfo.mockImplementation((nickname, profileUrl, cb) => {
    if (nickname) user.nickname = nickname;
    if (profileUrl) user.profileUrl = profileUrl;
    cb(user, null);
  });
  mockCreateApplicationUserListQuery.mockImplementation(() => ({
    hasNext: false
  }));
  mockCreateBlockedUserListQuery.mockImplementation(() => ({
    hasNext: false
  }));
  mockConnect.mockImplementation((userId, cb) => cb(true, null));
  mockCreateOpenChannel.mockImplementation((channelName, coverUrl, data, cb) => {
    channel = {
      ...channel,
      name: channelName
    };
    cb(channel, null);
  });
  mockGetOpenChannel.mockImplementation((channelUrl, cb) => {
    channel = {
      ...channel,
      channelUrl: channelUrl
    };
    cb(channel, null);
  });
  mockGetGroupChannel.mockImplementation((channelUrl, cb) => {
    channel = {
      ...channel,
      channelUrl: channelUrl
    };
    cb(channel, null);
  });
  mockCreateChannelWithUserIds.mockImplementation((inviteUserIdList, isDistinct, cb) => {
    cb(channel, null);
  });
  mockUnblockUserWithUserId.mockImplementation((unblockUserId, cb) => {
    cb(user, null);
  });
  mockBlockUserWithUserId.mockImplementation((blockUserId, cb) => {
    cb(user, null);
  });
  mockRemoveAllChannelHandlers.mockImplementation(() => {});
  mockChannelHandler.mockReset();
  mockConnectionHandler.mockReset();
  mockAddChannelHandler.mockReset();
  mockAddConnectionHandler.mockReset();
  mockRemoveChannelHandler.mockReset();
  mockCreateUserListQuery.mockImplementation(() => ({
    hasNext: true,
    next: cb => {
      cb();
    },
    limit: 2
  }));
}

export const mockUpdateCurrentUserInfo = jest.fn();
export const mockCreateApplicationUserListQuery = jest.fn();
export const mockCreateBlockedUserListQuery = jest.fn();
export const mockConnect = jest.fn();
export const mockCreateOpenChannel = jest.fn();
export const mockGetOpenChannel = jest.fn();
export const mockGetGroupChannel = jest.fn();
export const mockCreateChannelWithUserIds = jest.fn();
export const mockUnblockUserWithUserId = jest.fn();
export const mockBlockUserWithUserId = jest.fn();
export const mockRemoveAllChannelHandlers = jest.fn();
export const mockChannelHandler = jest.fn();
export const mockConnectionHandler = jest.fn();
export const mockAddChannelHandler = jest.fn();
export const mockAddConnectionHandler = jest.fn();
export const mockRemoveChannelHandler = jest.fn();
export const mockCreateUserListQuery = jest.fn(() => ({
  hasNext: true,
  next: cb => {
    cb();
  },
  limit: 2
}));

restoreAllDefaultMocks();

const methods = {
  getInstance: jest.fn(() => ({
    createApplicationUserListQuery: mockCreateApplicationUserListQuery,
    createBlockedUserListQuery: mockCreateBlockedUserListQuery,
    updateCurrentUserInfo: mockUpdateCurrentUserInfo,
    currentUser: user,
    disconnect: jest.fn(cb => {
      cb();
    }),
    OpenChannel: {
      createChannel: mockCreateOpenChannel,
      getChannel: mockGetOpenChannel
    },
    GroupChannel: {
      getChannel: mockGetGroupChannel,
      createChannelWithUserIds: mockCreateChannelWithUserIds
    },
    unblockUserWithUserId: mockUnblockUserWithUserId,
    blockUserWithUserId: mockBlockUserWithUserId,
    removeAllChannelHandlers: mockRemoveAllChannelHandlers,
    ChannelHandler: mockChannelHandler,
    ConnectionHandler: mockConnectionHandler,
    addChannelHandler: mockAddChannelHandler,
    addConnectionHandler: mockAddConnectionHandler,
    removeChannelHandler: mockRemoveChannelHandler,
    connect: mockConnect,
    createUserListQuery: mockCreateUserListQuery
  })),
  connect: mockConnect
};

/* eslint-disable no-undef */
jest.mock('sendbird', () => {
  const mock = jest.fn().mockImplementation(() => {
    return methods;
  });
  mock.getInstance = methods.getInstance;
  return mock;
});
