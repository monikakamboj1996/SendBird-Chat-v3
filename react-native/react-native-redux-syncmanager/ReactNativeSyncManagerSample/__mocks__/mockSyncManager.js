/* eslint-disable no-undef */
const collectionHandlerMocks = {
  onMessageEvent: jest.fn()
};

const messageCollectionMocks = {
  setCollectionHandler: jest.fn(),
  fetchSucceededMessages: jest.fn(),
  fetchFailedMessages: jest.fn()
};

const methods = {
  clearCache: jest.fn()
};

jest.mock('sendbird-syncmanager', () => {
  const mock = jest.fn().mockImplementation(() => {
    return methods;
  });
  mock.setup = jest.fn();
  mock.Options = function() {
    this.messageCollectionCapacity = 2000;
    this.messageResendPolicy = '';
  };
  mock.MessageCollection = function() {
    return messageCollectionMocks;
  };
  mock.MessageCollection.CollectionHandler = function() {
    return collectionHandlerMocks;
  };
  return mock;
});
