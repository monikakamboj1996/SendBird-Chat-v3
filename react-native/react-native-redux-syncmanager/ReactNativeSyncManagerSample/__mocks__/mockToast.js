export const Toast = {
  show: jest.fn(),
  durations: {
    SHORT: 'short'
  },
  positions: {
    BOTTOM: 'bottom'
  }
};

jest.mock('react-native-root-toast', () => {
  return Toast;
});
