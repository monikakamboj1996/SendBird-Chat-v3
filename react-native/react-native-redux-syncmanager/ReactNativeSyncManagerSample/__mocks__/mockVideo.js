jest.mock('react-native-video', () => {
  return 'Video';
});
