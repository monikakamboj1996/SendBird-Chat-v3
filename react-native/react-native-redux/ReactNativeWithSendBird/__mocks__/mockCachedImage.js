jest.mock('react-native-cached-image', () => ({
  CachedImage: 'CachedImage'
}));
