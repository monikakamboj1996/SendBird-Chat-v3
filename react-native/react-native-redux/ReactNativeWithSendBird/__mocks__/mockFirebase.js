/* eslint-disable no-undef */
jest.mock('react-native-firebase', () => ({
  messaging: jest.fn(() => ({
    hasPermission: jest.fn(() => Promise.resolve(true)),
    subscribeToTopic: jest.fn(),
    unsubscribeFromTopic: jest.fn(),
    requestPermission: jest.fn(() => Promise.resolve(true)),
    getToken: jest.fn(() => Promise.resolve('myMockToken')),
    onTokenRefresh: jest.fn(),
    onMessage: jest.fn()
  })),
  notifications: jest.fn(() => ({
    onNotification: jest.fn(),
    onNotificationDisplayed: jest.fn(),
    onNotificationOpened: jest.fn(),
    removeAllDeliveredNotifications: jest.fn()
  }))
}));
