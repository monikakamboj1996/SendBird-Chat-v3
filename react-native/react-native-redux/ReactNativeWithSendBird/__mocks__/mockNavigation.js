/* eslint-disable no-undef */
jest.mock('react-navigation', () => ({
  StackActions: {
    reset: jest.fn()
  },
  NavigationActions: {
    navigate: jest.fn()
  }
}));
