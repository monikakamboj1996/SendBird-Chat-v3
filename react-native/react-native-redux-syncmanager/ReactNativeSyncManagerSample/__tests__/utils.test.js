import {
  getChannelIndex,
  findChannelIndex,
  getMessageIndex,
  findMessageIndex,
  createChannelTitle,
  toast
} from '../src/utils';
import { Toast } from '../__mocks__/mockToast';

describe('utils', () => {
  it('getChannelIndex', () => {
    const newChannel = {
      url: '123'
    };
    const channels = [{ url: '456' }, { url: '756' }, { url: '901' }, { url: '123' }];
    expect(getChannelIndex(newChannel, channels)).toBe(3);
    expect(getChannelIndex(newChannel, channels.filter(channel => channel.url !== '123'))).toBe(-1);
  });

  it('findChannelIndex', () => {
    let newChannel = {
      url: '123',
      lastMessage: {
        createdAt: 8
      }
    };
    let channels = [
      {
        url: '456',
        lastMessage: {
          createdAt: 8
        }
      },
      {
        url: '756',
        lastMessage: {
          createdAt: 9
        }
      },
      {
        url: '901',
        lastMessage: {
          createdAt: 10
        }
      },
      {
        url: '123',
        lastMessage: {
          createdAt: 7
        }
      }
    ];
    expect(findChannelIndex(newChannel, channels)).toBe(3);
    expect(findChannelIndex(newChannel, channels)).toBe(3);
    newChannel = {
      url: '123',
      createdAt: 8
    };
    channels = [
      {
        url: '456',
        createdAt: 8
      },
      {
        url: '756',
        createdAt: 9
      },
      {
        url: '901',
        createdAt: 10
      },
      {
        url: '123',
        createdAt: 7
      }
    ];
    expect(findChannelIndex(newChannel, channels)).toBe(3);
    expect(findChannelIndex(newChannel, channels.filter(channel => channel.url !== '123'))).toBe(3);
    channels[1].createdAt = 5;
    expect(findChannelIndex(newChannel, channels)).toBe(1);
    newChannel.createdAt = 1;
    expect(findChannelIndex(newChannel, channels)).toBe(channels.length);
  });

  it('getMessageIndex', () => {
    const newMessage = {
      url: '123',
      isIdentical: msg => {
        return msg.url === '123';
      }
    };
    const messages = [{ url: '456' }, { url: '756' }, { url: '901' }, { url: '123' }];
    expect(getMessageIndex(newMessage, messages)).toBe(3);
    expect(getMessageIndex(newMessage, messages.filter(channel => channel.url !== '123'))).toBe(-1);
  });

  it('findMessageIndex createdAt', () => {
    const newMessage = {
      url: '123',
      createdAt: 5,
      reqId: 123
    };
    const messages = [{ createdAt: 6 }, { createdAt: 7 }, { createdAt: 8 }, { createdAt: 4 }];
    expect(findMessageIndex(newMessage, messages, true)).toBe(3);
    expect(findMessageIndex(newMessage, messages.filter(channel => channel.createdAt !== 4), true)).toBe(3);
  });

  it('findMessageIndex reqId', () => {
    const messages = [{ reqId: 6 }, { reqId: 7 }, { reqId: 8 }, { reqId: 4 }];
    expect(findMessageIndex({ reqId: 4 }, messages, true)).toBe(3);
    expect(findMessageIndex({ reqId: 7 }, messages, true)).toBe(1);
  });

  it('findMessageIndex messageId', () => {
    const messages = [{ messageId: 6 }, { messageId: 7 }, { messageId: 8 }, { messageId: 4 }];
    expect(findMessageIndex({ messageId: 4 }, messages, false)).toBe(3);
    expect(findMessageIndex({ messageId: 7 }, messages)).toBe(1);
  });

  it('createChannelTitle concise', () => {
    const channel = {
      members: [
        { nickname: '12345' },
        { nickname: '23456' },
        { nickname: '56789' },
        { nickname: '10123' },
        { nickname: '11234' }
      ]
    };
    expect(createChannelTitle(channel)).toMatchSnapshot();
  });

  it('createChannelTitle full', () => {
    const channel = {
      members: [{ nickname: '12345' }, { nickname: '23456' }, { nickname: '56789' }]
    };
    expect(createChannelTitle(channel)).toMatchSnapshot();
  });

  it('toast', () => {
    toast('hello');
    expect(Toast.show.mock.calls.length).toBe(1);
  });
});
