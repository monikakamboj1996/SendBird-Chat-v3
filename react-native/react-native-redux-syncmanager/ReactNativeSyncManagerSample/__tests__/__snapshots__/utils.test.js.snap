// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils createChannelTitle concise 1`] = `"12345, 23456, 567..."`;

exports[`utils createChannelTitle full 1`] = `"12345, 23456, 56789"`;
